### PR TITLE
MM-29918 Cypress/E2E: Add ability to run tests for EE and TE

### PR DESCRIPTION
**Pull request by saturninoabril**
#### Summary
This PR adds the ability to run test in CI for EE and TE.
- For EE: `CYPRESS_runWithEELicense=true node run_tests.js --stage='@prod'`, given that a license file can be found at `cypress/fixtures` folder.
- For TE: `node run_tests.js --stage='@prod' --exclude-group='@enterprise'`

Other changes included:
- For tests that require license to run:
  - Moved folder/files into `/enterprise` folder
  - Added each test file with `@enterprise` metadata

Note: ~~Test runs for EE and TE are ongoing. Will share report links once completed~~ See reports:
- EE test report - https://mm-cypress-report.s3.amazonaws.com/1159-ee-prod-master/mochawesome.html
  - failed tests known to master-prod
- TE test report - https://mm-cypress-report.s3.amazonaws.com/1158-te-prod-master/mochawesome.html
  - failed tests known to master-prod
  - other failed tests seemed to require a license. Will address separately to further verify license requirement.

#### Ticket Link
Jira ticket - https://mattermost.atlassian.net/browse/MM-29918

### CODE REVIEW

**Review by srkgupta — APPROVED** [review]
Thanks @saturninoabril. The changes LGTM.

File: e2e/run_tests.js [hunk 1]
```diff
@@ -35,6 +38,12 @@
  *      - will run all non-production tests
  * 4. "BROWSER='chrome' HEADLESS='false' node run_tests.js --stage='@prod' --group='@channel,@messaging'"
  *      - will run spec files matching stage and group values in Chrome (headed)
+ * 5. "CYPRESS_runWithEELicense=true node run_tests.js --stage='@prod'"
+ *      - will run all production tests
+ *      - typical test run for Enterprise Edition, given license file can be found in `cypress/fixtures` folder
+ * 6. "node run_tests.js --stage='@prod' --exclude-group'@enterprise'"
```
Comment (josephbaylon) [inline e2e/run_tests.js:44]:
```suggestion
 * 6. "node run_tests.js --stage='@prod' --exclude-group='@enterprise'"
```

**Review by josephbaylon — APPROVED** [review]
LGTM. Just minor missing `=` sign. Thanks @saturninoabril !!

File: e2e/run_tests.js (hunk 1 above)
Comment (saturninoabril) [reply e2e/run_tests.js:44]: Good catch, thanks!

**Review by prapti — APPROVED** [review]

**Pull request merged**